akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala

```diff
@@ -57,69 +64,202 @@ private[http] object Handshake {
             list of values indicating which extensions the client would like
             to speak.  The interpretation of this header field is discussed
             in Section 9.1.
-   */
-  def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
-    def find[T <: HttpHeader: ClassTag]: Option[T] =
-      headers.collectFirst {
-        case t: T ⇒ t
+    */
+    def isWebsocketUpgrade(headers: List[HttpHeader], hostHeaderPresent: Boolean): Option[UpgradeToWebsocket] = {
+      def find[T <: HttpHeader: ClassTag]: Option[T] =
+        headers.collectFirst {
+          case t: T ⇒ t
+        }
+
+      // val host = find[Host]
+      val upgrade = find[Upgrade]
+      val connection = find[Connection]
+      val key = find[`Sec-WebSocket-Key`]
+      val version = find[`Sec-WebSocket-Version`]
+      // val origin = find[Origin]
```

Comment:
Yes, it is important, but it is optional for Websocket, so we don't handle it on the lower layer. If you want to restrict origins you can do so from routing.